### ISSUE TITLE
[1.4.0.009]Level lost on world change.

### ISSUE THREAD

**Issue by InquisitorCM** [Mithion/ArsMagica2#1471]
Changelog say's, dat issuse have been fixed at .009 version, but i still have. 
When you change world(portal\teleport command) level turns into 0. End\nether\any world from divinerpg\thaumcraft - it does not matter.

**Comment by jjtParadox** [comment]
What mods are you using, and are you playing singleplayer or multiplayer?

**Comment by yeyifangg** [comment]
that maybe cause by the "PlayerTracker.onPlayerChangedDimension"
it is a forge API
which means any changeworldevent fired by bukkit plugins will not trigger this method
thus,you will lost your power level will

**Comment by jjtParadox** [comment]
If you're using a forge+bukkit combination, I did attempt to suggest a solution over [here](https://gitlab.prok.pw/KCauldron/KCauldron/issues/196), but it seems that there's an incompatibility between forge and bukkit concerning dimension transfers and respawns that cannot be fixed. In the perfect world every forge+bukkit combination would work just fine with all forge mods, but that's not the case here.

**Comment by jjtParadox** [comment]
Actually, looks like some changes have been made to that fork of KCauldron that I didn't notice that may have fixed the issue. What forge+bukkit combination are you using, if you are using one?

**Comment by InquisitorCM** [comment]
Multiplayer, core - Kcauldron 1.7.10-1614.201
Mods
Witchery
Thaumcraft
ThaumicTinker
Divinerpg
ArsMagica
IronChest

**Comment by jjtParadox** [comment]
Yeah it's KCaudron causing the issue. You should probably open a bug report with that group, as Ars Magica is using a methodology that works just fine in Forge, but breaks whenever KCauldron or similar is used. Feel free to reference back to this issue (and me) as well as to that issue I linked above (Issues #1209 and #1310 may also be of use).

If that doesn't work, try [this](https://github.com/TCPR/Thermos) fork of KCauldron, as they may have fixed the issue on their end.

**Comment by LemADEC** [comment]
Issue still happens randomly on that fork.

**Comment by InquisitorCM** [comment]
Using last Thermos build fixed this issuse.
